plugins/reach/plugins/ReachInternal/batch/KAsyncReachQueueHandler.class.php

```diff
@@ -0,0 +1,133 @@
+
+        $this->reachPlugin = KalturaReachClientPlugin::get(KBatchBase::$kClient);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function run($jobs = null)
+    {
+        //retrieve all entry vendor tasks using the response profile.
+        KBatchBase::impersonate(KBatchBase::$taskConfig->params->reachInternalVendorPartner);
+
+        //init filter
+        $filter = new KalturaEntryVendorTaskFilter();
+        $pager = new KalturaFilterPager();
+        $pager->pageSize = 100;
+
+        //Set response profile in order to add the vendor catalog item and/or reach profile to the entry vendor task response.
+        $responseProfile = $this->constructResponseProfile();
+        KBatchBase::$kClient->setResponseProfile($responseProfile);
+
+        $response = $this->reachPlugin->entryVendorTask->getJobs($filter, $pager);
+
+        if ($response->totalCount == 0)
+        {
+            KalturaLog::info('No jobs found to handle at this time. Exiting');
+            KBatchBase::unimpersonate();
+            return;
+        }
+
+        $handledTasksCounter = 0;
+        while ($handledTasksCounter < KBatchBase::$taskConfig->params->taskHandleLimit) {
+            foreach ($response->objects as $entryVendorTask) {
+                /* @var $entryVendorTask KalturaEntryVendorTask */
+
+                //retrieve associated catalog item and retrieve appropriate engine based on the engineType property
+                $catalogItem = $entryVendorTask->relatedObjects[0]->objects[0];
+                /* @var $catalogItem KalturaVendorCatalogItem */
+                $engine = KReachVendorTaskProcessorEngine::getInstance($catalogItem->engineType);
+                if (!$engine)
+                {
+                    KalturaLog::info('No engine type found to process entry vendor task ID: ' . $entryVendorTask->id);
+                    continue;
+                }
+
+                try {
+                    $engine->moveTaskToStatus($entryVendorTask, KalturaEntryVendorTaskStatus::PROCESSING);
+                    $engine->handleTask($entryVendorTask);
+                    KBatchBase::unimpersonate();
+                } catch (Exception $e) {
+                    KalturaLog::err('Error occurred processing vendor task with ID ' . $entryVendorTask->id . '. Message: ' . $e->getMessage());
+                    KBatchBase::unimpersonate();
+                }
+
+                $handledTasksCounter++;
+                if ($handledTasksCounter >= KBatchBase::$taskConfig->params->taskHandleLimit)
+                {
+                    KalturaLog::info('Entry vendor task processing limit reached for current run. Exiting.');
+                    break;
+                }
+            }
+
+            KBatchBase::impersonate(KBatchBase::$taskConfig->params->reachInternalVendorPartner);
+            KBatchBase::$kClient->setResponseProfile($responseProfile);
+            $response = $this->reachPlugin->entryVendorTask->getJobs($filter, $pager);
```

Comment:
if response return empty - you will fail here, anyway better change thsi loop to 
do 
  getJobs
  validate Jobs or break
while counter<max